### ISSUE TITLE
Fix URI validation

### ISSUE THREAD

**Issue by KellerFuchs** [Queertoo/Queerlink#2]
The current code rejects perfectly valid URLs, due to `uri.host != nil`.

More generally, Elixir's parser [is not validating](https://github.com/elixir-lang/elixir/issues/198#issuecomment-68028223).  I will investigate using [ex_uri](https://github.com/heroku/ex_uri) for that purpose.
@Annwenn Would [ex_uri](https://github.com/heroku/ex_uri) be an acceptable dependency to pull in?

**Comment by ghost** [comment]
> due to uri.host != nil

Yes, so people don't fill the database with garbage. But alas it also rejects aliased domains.

> I will investigate using ex_uri for that purpose

Yeah of course, poke around with `ex_uri`

> Would ex_uri be an acceptable dependency to pull in?

Once the aforementioned poking will produce results, we'll think about that (I'm not fundamentally opposed, though)

**Comment by KellerFuchs** [comment]
> > due to uri.host != nil
> 
> Yes, so people don't fill the database with garbage. But alas it also rejects aliased domains.

Yes, I didn't say we should not do validation at all.

The specific case that made me investigate was that I tried to shorten a data URI  =^.^'=
